src/main/java/org/junitpioneer/jupiter/resource/PathDeleter.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2016-2022 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junitpioneer.jupiter.resource;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+
+// visible for testing purposes
+class PathDeleter extends SimpleFileVisitor<Path> {
+
+	static final PathDeleter INSTANCE = new PathDeleter();
```

Comment:
Candidate for the [Enum singleton pattern](https://java-design-patterns.com/patterns/singleton/).

Reply:
I agree! But unfortunately, IntelliJ tells me that "No extends clause [is] allowed for enum". 😭